pkg/ddc/vineyard/transform.go

```diff
@@ -0,0 +1,257 @@
+
+	value.Fuse.CleanPolicy = runtime.Spec.Fuse.CleanPolicy
+
+	value.Fuse.NodeSelector = e.transformFuseNodeSelector(runtime)
+
+	value.Fuse.TargetPath = e.getMountPoint()
+	e.transformResourcesForFuse(runtime, value)
+}
+
+func (e *VineyardEngine) transformMasterSelector(runtime *datav1alpha1.VineyardRuntime) map[string]string {
+	properties := map[string]string{}
+	if runtime.Spec.Master.NodeSelector != nil {
+		properties = runtime.Spec.Master.NodeSelector
+	}
+	return properties
+}
+
+func (e *VineyardEngine) transformMasterPorts(runtime *datav1alpha1.VineyardRuntime) map[string]int {
+	ports := map[string]int{
+		MasterClientName: MasterClientPort,
+		MasterPeerName:   MasterPeerPort,
+	}
+	if len(runtime.Spec.Master.Ports) > 0 {
+		for key, value := range runtime.Spec.Master.Ports {
+			ports[key] = value
+		}
+	}
+	return ports
+}
+
+func (e *VineyardEngine) transformMasterOptions(runtime *datav1alpha1.VineyardRuntime) map[string]string {
+	options := map[string]string{
+		WorkerReserveMemory: DefaultWorkerReserveMemoryValue,
+		WorkerEtcdPrefix:    DefaultWorkerEtcdPrefixValue,
+	}
+	if len(runtime.Spec.Master.Options) > 0 {
+		for key, value := range runtime.Spec.Master.Options {
+			options[key] = value
+		}
+	}
+	return options
+}
+
+func (e *VineyardEngine) transformWorkerOptions(runtime *datav1alpha1.VineyardRuntime) map[string]string {
+	options := map[string]string{}
+	if len(runtime.Spec.Worker.Options) > 0 {
+		options = runtime.Spec.Worker.Options
+	}
+
+	return options
+}
+
+func (e *VineyardEngine) transformWorkerPorts(runtime *datav1alpha1.VineyardRuntime) map[string]int {
+	ports := map[string]int{
+		WorkerRPCName:      WorkerRPCPort,
+		WorkerExporterName: WorkerExporterPort,
+	}
+	if len(runtime.Spec.Worker.Ports) > 0 {
+		for key, value := range runtime.Spec.Worker.Ports {
+			ports[key] = value
+		}
+	}
+	return ports
+}
+
+func (e *VineyardEngine) transformFuseNodeSelector(runtime *datav1alpha1.VineyardRuntime) map[string]string {
+	nodeSelector := map[string]string{}
+	nodeSelector[e.getFuseLabelName()] = "true"
+	return nodeSelector
+}
+
+func (e *VineyardEngine) transformTieredStore(runtime *datav1alpha1.VineyardRuntime) TieredStore {
+	quota := resource.MustParse("4Gi")
+	tieredStore := TieredStore{
+		Levels: []Level{
+			{
+				MediumType: "MEM",
+				Level:      0,
+				Quota:      &quota,
+			},
+		},
+	}
+	if len(runtime.Spec.TieredStore.Levels) != 0 {
+		tieredStore = TieredStore{}
+		for _, level := range runtime.Spec.TieredStore.Levels {
+			if level.MediumType == "MEM" {
+				tieredStore.Levels = append(tieredStore.Levels, Level{
+					MediumType: level.MediumType,
+					Quota:      level.Quota,
+				})
+			} else {
+				tieredStore.Levels = append(tieredStore.Levels, Level{
+					MediumType:   level.MediumType,
+					VolumeType:   level.VolumeType,
+					VolumeSource: level.VolumeSource,
+					Path:         level.Path,
```

Comment:
Reading the helm function `vineyard.spill.path`, I think `Level=1` should be set here?

Reply:
Thanks for the catching.